libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -1677,6 +1677,12 @@ class MapboxNavigation @VisibleForTesting internal constructor(
      * Invoke when any component of EV data is changed so that it can be used in refresh requests.
      * You can pass only changed components of EV data via [data], all the previous values
      * that have not changed will be cached on the SDK side.
+     * NOTE: accepted EV refresh parameters are:
+     * 1) "energy_consumption_curve";
+     * 2) "ev_initial_charge";
+     * 3) "auxiliary_consumption";
+     * 4) "ev_pre_conditioning_time".
+     * If you pass any other parameters via this method, the route refresh request will fail.
```

Comment:
```suggestion
     *
     * **NOTE: Only provide parameters that are compatible with an EV route refresh. If you pass any other parameters via this method, the route refresh request will fail.**
     *
```
Let's not bake in any experimental API assumptions into code/docs. Which parameters are support should be shared in the Nav API docs or directly with the customer.

Reply:
In the refresh case we'll need to work directly with the customers. Once the EV feature becomes stable, we'll build a documented, type-safe strategy.